model/src/main/java/org/projectnessie/model/Namespace.java

```diff
@@ -86,16 +86,25 @@ public boolean isEmpty() {
    */
   public static Namespace of(String... elements) {
     Objects.requireNonNull(elements, "elements must be non-null");
-    if (elements.length == 0 || "".equals(elements[0])) {
+    if (elements.length == 0 || (elements.length == 1 && "".equals(elements[0]))) {
```

Comment:
when does this usually get called with an array of length 1 containing just the empty string? it looks odd

Reply:
this is mainly for the `Namespace.of(""))` case and without this check `Namespace.of("", "something")` will produce `Namespace.EMPTY`